### ISSUE TITLE
Consolidate DB calls for fetching user tags

### ISSUE THREAD

**Issue by adamalton** [web3-storage/web3.storage#1383]
There are a few inefficiencies in the way that we're currently fetching user and user tag data from the DB in the `userAccountGet` (`/user/account`) and `userInfoGet` (`/user/info`) endpoints.

In #1338 we added fetching of the user tag value for the storage limit into [userAccountGet](https://github.com/web3-storage/web3.storage/blob/63ec0b2340d247016faee4f69b504437cd29317a/packages/api/src/user.js#L115-L116). In #1322 fetching of user tags was added to [userInfoGet](https://github.com/web3-storage/web3.storage/blob/16b4309c269938ba9c1644f00c6b92f7392dd29c/packages/api/src/user.js#L128).

These two PRs happened in parallel, but now that they're both merged, we could consolidate the fetching of the user tags by removing the tag fetching from `userAccountGet` and making the front-end use the value that's returned by `userInfoGet` instead.

But there's actually another inefficiency here as well. In both endpoints, the `withMagicToken` middleware has already fetched both the user and the user tags and attached them to `request.auth.user`! So `userAccountGet` is making a totally unnecessary DB call for the storage quota tag, and `userInfoGet` is making totally unnecessary DB calls for both the user and the user tags.

So we can definitely rip some code out here.

The way that `withMagicToken` fetches the user tags isn't very efficient though (it makes a separate query for them). The way that `db.DBClient.getUser` does it (with a JOIN) is better, but still not great because it's not filtering by `deleted_at`. The way that `db.DBClient.getUserTagValue` nicely filters by `deleted_at`, but requires you to have already fetched the user in a separate query. We can make a best-of-all-worlds solution here!

**Comment by drewdelano** [comment]
I can definitely collapse the two calls into one, but I think we still need a call to get the user's proposal tags since doesn't seem to be included in magic auth:

![image](https://user-images.githubusercontent.com/22156330/174903926-12d911c8-1382-4231-ae06-23541bd34696.png)

![image](https://user-images.githubusercontent.com/22156330/174903939-516f9f06-601a-42e3-9028-daac7d3a08e0.png)

**Comment by drewdelano** [comment]
does that sound right to you @adamalton ?

**Comment by JeffLowe** [comment]
@jsdevel Mind taking a look at this?

**Comment by jsdevel** [comment]
you're correct @drewdelano .  proposed user tags are tags the user is proposing an admin add to their account, and they're stored in a separate table.  not really needed for auth.

**Comment by drewdelano** [comment]
cool, i'll go ahead and continue collapsing the two calls but continue asking for the proposal tags separately next week

**Comment by adamalton** [comment]
Ah, the user tag proposals didn't exist when I wrote this ticket. That makes things a bit trickier to optimise.

Perhaps in the `userInfoGet` endpoint, rather than re-fetching the user (which is already attached to `request.auth.user`) and the user tags (as well as the user tag proposals), we could just make a simpler query to fetch the user tag proposals, which is the only bit that we don't have at that point?

I feel like there are still one or two optimisations that we could make among all this, but it's maybe slightly less straightforward than it was when I wrote the ticket. If you want to take it on then I'll leave you to explore the possibilities.
